### PR TITLE
feat: add support for .NET SDK V4

### PR DESCRIPTION
**Pull request by 96malhar**
*Issue #, if available:*
DOTNET-7761

*Description of changes:*
* Updated the .NET SDK dependencies to the latest version 4.0.0-preview.4
* Marked project as trimmable
* Added SourceLink support

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by normj — APPROVED** [review]

File: src/AWS.DistributedCacheProvider/AWS.DistributedCacheProvider.csproj [hunk 1]
```diff
@@ -35,8 +44,8 @@
 		<PackageReference Include="Microsoft.Extensions.Caching.Abstractions" Version="6.0.0" />
 		<PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="6.0.0" />
 		<PackageReference Include="Microsoft.Extensions.Options" Version="6.0.0" />
-		<PackageReference Include="AWSSDK.DynamoDBv2" Version="3.7.302.15" />
-		<PackageReference Include="AWSSDK.Extensions.NETCore.Setup" Version="3.7.300" />
+		<PackageReference Include="AWSSDK.DynamoDBv2" Version="4.0.0-preview.4" />
+		<PackageReference Include="AWSSDK.Extensions.NETCore.Setup" Version="4.0.0-preview.4" />
```
Comment (philasmar) [inline src/AWS.DistributedCacheProvider/AWS.DistributedCacheProvider.csproj:48]: You are missing a reference to ```<PackageReference Include="Microsoft.SourceLink.GitHub" Version="8.0.0" PrivateAssets="All" />```. Without it, SourceLink will not work.

**Review by philasmar — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -39,6 +39,42 @@ To send us a pull request, please:
...
+```
+The `change file` lists all the modified projects, the changelog message for each project as well as the increment type. 
+These files are located in the repo at .autover/changes/
+You can use the `AutoVer` tool to create the change file. You can install it using the following command:
+```
+dotnet tool install -g AutoVer
+```
+You can create the `change file` using the following command:
+```
+autover change --project-name "Amazon.AspNetCore.Identity.Cognito" -m "Fixed an issue causing a failure somewhere
```
[the remark this answers is not in the document]
Comment (96malhar) [reply CONTRIBUTING.md:66]: Done

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -39,6 +39,42 @@ To send us a pull request, please:
...
+```
+Note: Make sure to run the command from the root of the repository.
+You can update the command to specify which project you are updating.
+The available projects are:
+* Amazon.AspNetCore.Identity.Cognito
```
[the remark this answers is not in the document]
Comment (96malhar) [reply CONTRIBUTING.md:71]: Done

File: src/AWS.DistributedCacheProvider/AWS.DistributedCacheProvider.csproj [hunk 4]
```diff
@@ -1,11 +1,11 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
 	<PropertyGroup>
-		<TargetFramework>net6.0</TargetFramework>
+		<TargetFrameworks>net6.0;net8.0</TargetFrameworks>
 		<RootNamespace>AWS.DistributedCacheProvider</RootNamespace>
 		<ImplicitUsings>enable</ImplicitUsings>
 		<Nullable>enable</Nullable>
-		<Version>0.9.3</Version>
+		<Version>4.0.0-preview.1</Version>
```
[the remark this answers is not in the document]
Comment (96malhar) [reply src/AWS.DistributedCacheProvider/AWS.DistributedCacheProvider.csproj:8]: Changed the version to `2.0.0-preview.1`

File: src/AWS.DistributedCacheProvider/AWS.DistributedCacheProvider.csproj (hunk 1 above)
Comment (96malhar) [reply src/AWS.DistributedCacheProvider/AWS.DistributedCacheProvider.csproj:48]: Done

**Review by philasmar — APPROVED** [review]

**Pull request merged**